stripe/src/main/java/com/stripe/android/model/SetupIntent.kt

```diff
@@ -206,12 +206,16 @@ data class SetupIntent internal constructor(
     }
 
     companion object {
+        private const val clientSecretRegexPattern = "^seti_[^_]+_secret_[^_]+$"
 
-        @JvmStatic
         fun fromJson(jsonObject: JSONObject?): SetupIntent? {
             return jsonObject?.let {
                 SetupIntentJsonParser().parse(it)
             }
         }
+
+        fun isClientSecretValid(clientSecret: String): Boolean {
+            return Pattern.compile(clientSecretRegexPattern).matcher(clientSecret).matches()
+        }
```

Comment:
Intentionally public?

Reply:
Same, I think it's fine to be public?

Reply:
I'm fine with these being public, but I'd err on the side of leaving them internal unless there's a good reason to make them public. You can always make them public in a future PR.